internal/templates/render/render.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+type Command struct {
+	cli.BaseCommand
+
+	template        string
+	spec            string
+	dest            string
+	gitProtocol     string
+	logLevel        string
+	allowNonGitDest bool
+	forceOverwrite  bool
+	keepTempDirs    bool
+	inputs          map[string]string
+}
+
+// Desc implements cli.Command.
+func (c *Command) Desc() string {
+	return "Instantiate a template to setup a new app or add config files"
+}
+
+func (c *Command) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  The {{ COMMAND }} command renders the given template into the given dest
+  directory.
+`
+}
+
+func (c *Command) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	sect := set.NewSection("Render options")
+	sect.StringVar(&cli.StringVar{
+		Name:    "template",
```

Comment:
Should this be `name`? Considering we're already under `abc templates`

Reply:
I see what you mean. "name" doesn't seem quite right though, because it's not just a name, it could be a URL or a file path or whatever. Maybe "location"? I'm leaning toward just leaving it as "template".

Reply:
Or maybe `--source` or `-s` which maps to `--dest`. It's a little itchy to me to repeat "template"

Reply:
"source" is great, changed to that.